### DISCUSSION TITLE
found multiple images named 'Im0'

### DISCUSSION THREAD

**Discussion by Fake45mar** [pymupdf/PyMuPDF#2332] [Q&A] [answered]
## Overview
Name of all images extracted on page are the same.
https://pymupdf.readthedocs.io/en/latest/page.html#Page.get_images - item[7] == 'Im0' in all cases
https://pymupdf.readthedocs.io/en/latest/document.html#Document.get_page_images

## Repro
image_list = page.get_images()
image_list: [(519, 0, 1162, 1642, 8, 'ICCBased', '', 'Im0', 'DCTDecode'), (544, 0, 583, 1642, 8, 'ICCBased', '', 'Im0', 'DCTDecode'), (595, 594, 144, 80, 8, 'DeviceGray', '', 'Im0', 'DCTDecode')]

## Expected behavior
image_list = page.get_images()
image_list: [(519, 0, 1162, 1642, 8, 'ICCBased', '', 'Im0', 'DCTDecode'), (544, 0, 583, 1642, 8, 'ICCBased', '', 'Im1', 'DCTDecode'), (595, 594, 144, 80, 8, 'DeviceGray', '', 'Im2', 'DCTDecode')]

## Screenshots 
![image](https://user-images.githubusercontent.com/38050410/230911803-ca50cf9c-d278-440d-b01d-e45e39f9e77a.png)

## Your configuration (mandatory)
 - macOS Monterey 12.5.1
 - Python version 3.9.9
 - pip3 install wheel
    brew install swig
    brew install freetype
    pip3 install PyMuPDF==1.21.1

**Comment by JorjMcKie** [comment]
This has nothing to do with (Py-) MuPDF, but is specified that way by the PDF creator.

**Reply by Fake45mar** [reply]
I can confirm that it's data oriented bug. I ran a batch of documents and not on each got this issue.
Ok, is it any option to obtain image bbox not by the name pdf creator specified?

**Reply by JorjMcKie** [reply, marked as answer]
If the same image reference name occurs lime "Im0", then it at least should only happen if the image references occur in different XObjects. You can reveal this situation by `page.get_images(True)`. The items of this list are longer and contain the xref of the responsible XObject. If it was the page itself, that last item is 0.
```python
In [1]: import fitz
In [2]: doc=fitz.open("x0.pdf")
In [3]: page=doc[0]
In [4]: page.get_images(True)
Out[4]: [(15, 16, 1200, 1200, 8, 'DeviceRGB', '', 'Im1', 'FlateDecode', 0)]  # image invoked by page itself!
```
